app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineRunErrorDetails.tsx

```diff
@@ -0,0 +1,338 @@
+        <a href={link} target="_blank">
+          {link} <LaunchIcon fontSize="small" />
+        </a>
+      );
+      nodes.push(chunks[i]);
+    }
+
+    return <pre>{nodes}</pre>;
+  }
+
+  if (currentRun?.status !== 'FAILED' || !errorDetails?.length || loading) {
+    return null;
+  }
+
+  return (
+    <ThemeWrapper>
+      <PipelineRunErrorDetailsWrapper>
+        <Provider store={PipelineMetricsStore}>
+          <ShortErrorMessage>
+            <PipelineErrorCountMessage classifiedErrorCount={errorDetails?.length} />
+            <Button
+              color="inherit"
+              onClick={toggleErrorDetails}
+              data-testid={getDataTestid(
+                detailsExpanded ? `${TEST_PREFIX}.closeButton` : `${TEST_PREFIX}.viewDetailsButton`
+              )}
+            >
+              {detailsExpanded
+                ? T.translate(`${PREFIX}.closeButton`)
+                : T.translate(`${PREFIX}.viewDetailsButton`)}
+            </Button>
+          </ShortErrorMessage>
+          {detailsExpanded && (
+            <ErrorDetailsContainer>
+              <Table>
+                <TableHead>
+                  <TableRow>
+                    <TableCell>{T.translate(`${PREFIX}.errorCategoryHeader`)}</TableCell>
+                    <TableCell classes={TdClassesOverride}>
+                      {T.translate(`${PREFIX}.errorMessageHeader`)}
+                    </TableCell>
+                    <TableCell>{T.translate(`${PREFIX}.errorReasonHeader`)}</TableCell>
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {errorDetails?.map((err) => (
+                    <TableRow>
+                      <TableCell>{err.errorCategory}</TableCell>
+                      <TableCell classes={TdClassesOverride}>
+                        {renderWithLink(err.errorMessage, err.supportedDocumentationUrl)}
+                      </TableCell>
+                      <TableCell>
+                        {renderWithLink(err.errorReason, err.supportedDocumentationUrl)}
+                      </TableCell>
+                    </TableRow>
+                  ))}
+                </TableBody>
+              </Table>
+              <ErrorImprovementMessage>
+                {improvementSurveyEnabled ? (
+                  <p>
+                    {/* TODO (CDAP-21109): Add i18n strings */}
+                    Help us improve the error classification. Raise improvement <a href="#">here</a>
+                    .
+                  </p>
+                ) : (
+                  <p />
+                )}
+                <LogsButtonsContaner>
+                  <Button
+                    color="secondary"
+                    href={getDownloadLogsUrl(dataFetcher)}
+                    endIcon={<GetAppIcon />}
+                    data-testid={getDataTestid(`${TEST_PREFIX}.downloadLogsButton`)}
+                  >
+                    {T.translate(`${PREFIX}.downloadLogsButton`)}
```

Comment:
Could you please update the screenshot in the PR's description reflecting latest changes?

Reply:
Updated the screenshot